components/bl-weather-component/src/index.js

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'react';
+
+import ReactWeather, { useOpenWeather } from './lib/reactWeather';
+
+const { cn } = BackendlessUI.CSSUtils;
+const UNITS = {
+  standard: { temperature: 'K', windSpeed: 'm/s' },
+  metric: { temperature: '°C', windSpeed: 'm/s' },
+  imperial: { temperature: '°F', windSpeed: 'mph' },
+};
+
+export default function WeatherComponent({ component, elRef, settings }) {
+  const { classList, style, display, currentLocation, city, coords, forecast, unit, theme, lang } = component;
+  const { API_KEY } = settings;
+
+  const [lat, setLat] = useState(null);
+  const [lon, setLon] = useState(null);
+  const [weatherCity, setWeatherCity] = useState(null);
+
+  useEffect(() => {
+    if (currentLocation && navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(function (position) {
+        setLat(position.coords.latitude);
+        setLon(position.coords.longitude);
+      })
+    } else if (city) {
+      fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${API_KEY}&units=metric`)
+        .then(response => response.json())
+        .then(data => {
+          setLat(data.coord.lat);
+          setLon(data.coord.lon);
+          setWeatherCity(data.name);
+        })
+    } else if (coords && coords.lat && coords.lng) {
+      setLat(coords.lat);
+      setLon(coords.lng);
+    }
+  }, [currentLocation, city, coords, API_KEY]);
+
+  useEffect(() => {
+    if (lat && lon && !weatherCity) {
+      fetchWeatherCity(lat, lon);
+    }
+  }, [lat, lon]);
+
+  const fetchWeatherCity = (lat, lon) => {
+    fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${API_KEY}&units=metric`)
+      .then(response => response.json())
+      .then(data => {
+        setWeatherCity(data.name);
+      })
+  };
```

Comment:
In my opinion, it is better to move all logic from 16 to 52 line into a custom hook.